libs/blocks/preflight/checks/preflightExecutor.js

```diff
@@ -0,0 +1,36 @@
+import { getPreflightResults } from './preflightApi.js';
+
+let preflightResults;
+
+const executePreflightChecks = async () => {
+  if (preflightResults) {
+    return preflightResults;
+  }
+
+  const url = window.location.pathname;
+  const area = document;
+
+  preflightResults = getPreflightResults(url, area).then((results) => {
+    preflightResults = results.runChecks;
+    return results.runChecks;
+  });
+
+  return preflightResults;
+};
+
+export function hasPreflightFailures(results = null) {
+  const cache = results || preflightResults;
```

Comment:
don't u already cache it in the preflight API, why here as well?
also why do you need a method to executePreflightChecks if the API already has a getResults method